### PR TITLE
fix the integration for routing admin with content bundle

### PR DESCRIPTION
**Pull request by dbu**
i realized that the content basepath is still configured on the routing bundle, but not used at all. then i realized we did not properly handle content base path here in the admin. this should fix all of that

### CODE REVIEW

File: src/DependencyInjection/CmfSonataPhpcrAdminIntegrationExtension.php [hunk 1]
```diff
@@ -65,6 +63,18 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('enhancer.xml');
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * Overwritten because configuration can not be auto instantiated as it has a constructor.
+     */
+    public function getConfiguration(array $config, ContainerBuilder $container)
```
Comment (dbu) [inline src/DependencyInjection/CmfSonataPhpcrAdminIntegrationExtension.php]: this is needed to make config:dump-reference work.

File: src/DependencyInjection/Factory/RoutingAdminFactory.php [hunk 2]
```diff
@@ -35,16 +35,22 @@ public function getKey()
...
         $builder->scalarNode('basepath')->defaultNull()->end();
+        $builder->scalarNode('content_basepath')->defaultNull()->end();
     }
 
     /**
      * {@inheritdoc}
      */
     public function create(array $config, ContainerBuilder $container, XmlFileLoader $loader)
     {
+        if (!$config['enabled']) {
```
Comment (dbu) [inline src/DependencyInjection/Factory/RoutingAdminFactory.php]: why would we want to load routing services if we are not enabled?

Comment (wouterj) [inline src/DependencyInjection/Factory/RoutingAdminFactory.php]: This should be reverted. `AdminFactory#create()` is not called when the factory is disabled: https://github.com/symfony-cmf/sonata-phpcr-admin-integration-bundle/blob/65036a3c48e214c78df9460b156ad052eff1d36b/src/DependencyInjection/CmfSonataPhpcrAdminIntegrationExtension.php#L81

Comment (dbu) [inline src/DependencyInjection/Factory/RoutingAdminFactory.php]: ah ok. fixed it. could we also check for enabled before calling process? that bit looks a bit too implicit for my taste (we check if one of the parameters that would be set if enabled is indeed set)

Comment (wouterj) [inline src/DependencyInjection/Factory/RoutingAdminFactory.php]: > (we check if one of the parameters that would be set if enabled is indeed set)

That's normal behaviour for compiler passes. Imo, we should keep it. Sometimes, the factory is enabled but a specific feature isn't, so the compiler pass shouldn't be executed, etc. Let's keep all execution logic in the pass itself.

**Pull request merged**